lib/transactional.pm

```diff
@@ -147,7 +148,27 @@ sub check_reboot_changes {
     process_reboot(trigger => 1) if $change_happened;
 }
 
-
+# Check slem target version after migration
+sub check_slem_target_version {
+    my $release = script_output "cat /etc/os-release";
+    my %checker = ();
```

Comment:
Please simplify this code to just check the version of the product, with one or two values would be enough, you don't need to match the whole file, probably this validation would easily break. As mentioned above this is not multi-product so it will not work and will end up in conditional code.
Additionally the record_info way to show the information and the way to give the error is not useful, you need to use libraries that check expected value versus current and show you information in one shot if possible not through multiple boxes in the screen.

My recommendation is that you pick one single value for version and check it (with library or without library with a proper message) without loops and complications.

Reply:
Updated.